ls1intum/Artemis
# Ensure that users only receive notifications for their courses

### Checklist

### Motivation and Context
Some users receive recent notifications for courses to which they do not belong.

### Description
The REST route `/notifications` related to `findAllExceptSystem()` in the `NotficationService` seems to work as the *unwanted* notifications are not displayed there, only in the `notification-container.component`. 

Therefore I adapted the implementation of `findAllRecentExceptSystem()` in the `NotficationService` that is related to the REST route `/notifications/for-user` as closely as possible to the implementation of `findAllExceptSystem()`.

I also added several test cases for the above mentioned rest routes.

Additionally, I made sure that the notification service gets cleaned up when the user logs out.

### Steps for Testing
1. Log in to Artemis
2. Navigate to Course Administration
3. Click on the notification bell in the top navigation
4. Check that the displayed notifications only belong to courses of the current user
